keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,140 @@
+            lower and upper bounds for the aspect ratio of the cropped image
+            before resizing it to `target_size`. Defaults to (3./4., 4./3.).
+        area_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. Represents the lower and upper bounds for
+            the area relative to the original image of the cropped image before
+            resizing it to `target_size`. Defaults to (0.08, 1.0).
+        interpolation: interpolation method used in the `ImageProjectiveTransformV3` op.
+             Supported values are `"nearest"` and `"bilinear"`.
+             Defaults to `"bilinear"`.
+        fill_mode: fill_mode in the `ImageProjectiveTransformV3` op.
+             Supported values are `"reflect"`, `"wrap"`, `"constant"`, and `"nearest"`.
+             Defaults to `"reflect"`.
+        fill_value: fill_value in the `ImageProjectiveTransformV3` op.
+             A `Tensor` of type `float32`. The value to be filled when fill_mode is
+             constant".  Defaults to `0.0`.
+        seed: Integer. Used to create a random seed.
+    """
+    def __init__(self,
+                 target_size,
+                 aspect_ratio_factor=(3. / 4., 4. / 3.),
+                 area_factor=(0.08, 1.0),
+                 interpolation="bilinear",
+                 fill_mode="reflect",
+                 fill_value=0.0,
+                 seed=None,
+                 **kwargs):
+        super(RandomResizedCrop, self).__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+        self.aspect_ratio_factor = aspect_ratio_factor
+        self.area_factor = preprocessing.parse_factor(area_factor,
+                                                      param_name="area_factor",
+                                                      seed=seed)
+
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.seed = seed
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
+            warnings.warn(
+                "RandomResizedCrop received both `area_factor=0.0` and "
+                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
+                "augmentation.")
+
+    def get_random_transformation(self,
+                                  image=None,
+                                  label=None,
+                                  bounding_box=None):
+        area_factor = self.area_factor()
+        aspect_ratio = tf.random.uniform((),
+                                         minval=self.aspect_ratio_factor[0],
+                                         maxval=self.aspect_ratio_factor[1],
+                                         dtype=tf.float32)
+
+        new_height = tf.clip_by_value(
+            tf.sqrt(area_factor / aspect_ratio), 0.0,
+            1.0)  # to avoid unwanted/unintuitive effects
+        new_width = tf.clip_by_value(tf.sqrt(area_factor * aspect_ratio), 0.0,
```

Comment:
just so I understand, what would be unintuitive about making the new with bigger than 1.0?   Is there a reason we crop first then resize instead of resizing first then cropping?

You could resize the image by 0.5, then randomly crop a 10x10 section.

OR resize an image by 2.0, then crop to 10x10.

Does this explain why I think we can support area_factor > 1.0?

Reply:
I'm not sure how that would make a difference, since area_factor and aspect_ratio are sampled randomly. Since the aforementioned "10x10" is also calculated on the fly using random values, it would not matter in a distribution of images  whether we resize first or not. Please correct me if I'm wrong.

Reply:
intuitively I would think area_factor would resize the image, then aspect ratio fact would distort the image, then the crop operation would happen last.

I guess this is not how it is done in the rest of the ML world?

Reply:
No I believe the convention I told is generally followed, at least that's what I'm familiar with. @sayakpaul might have a better idea of this.